tests/e2e/performance/test_small_file_workload.py

```diff
@@ -125,16 +429,50 @@ def test_smallfile_workload(self, ripsaw, file_size, files, threads, samples, in
         )
         start_time = time.time()
         timeout = 1800
+
+        # Getting the UUID from inside the benchmark pod
+        output = bench_pod.exec_oc_cmd(f'exec {small_file_client_pod} env')
+        for line in output.split():
+            if 'uuid=' in line:
+                uuid = line.split('=')[1]
+        log.info(f'the UUID of the test is : {uuid}')
+        full_results = SmallFileResultsAnalyse(uuid, sf_data)
+
+        # Initialaize the results doc file.
+        full_results.add_key('user', sf_data['spec']['test_user'])
+        full_results.add_key('ocp_version', get_ocp_version())
+        full_results.add_key('ocp_build', get_build())
+        full_results.add_key('ocp_channel', get_ocp_channel())
+        full_results.add_key('ocs_version', get_ocs_version())
+        full_results.add_key('vendor', get_provider())
+        full_results.add_key('start_time',
+                             time.strftime('%Y-%m-%dT%H:%M:%SGMT',
+                                           time.gmtime()))
+        full_results.add_key('global_options', {
+            'files': files,
+            'file_size': file_size,
+            'storageclass': sf_data['spec']['workload']['args']['storageclass'],
+            'vol_size': sf_data['spec']['workload']['args']['storagesize']
+        })
+
         while True:
             logs = bench_pod.exec_oc_cmd(
                 f'logs {small_file_client_pod}',
                 out_yaml_format=False
             )
             if "RUN STATUS DONE" in logs:
-                log.info("SmallFile Benchmark Completed Successfully")
+                full_results.add_key('end_time',
+                                     time.strftime('%Y-%m-%dT%H:%M:%SGMT',
+                                                   time.gmtime()))
+                full_results.read()
+                full_results.add_key('hosts', full_results.get_clients_list())
+                full_results.init_full_results()
+                full_results.aggregate_host_results()
+                test_status = full_results.aggregate_samples_results()
+                full_results.write()
                 break
 
             if timeout < (time.time() - start_time):
-                raise TimeoutError(f"Timed out waiting for benchmark to complete")
+                raise TimeoutError("Timed out waiting for benchmark to complete")
             time.sleep(30)
-        assert not get_logs_with_errors()
+        assert (not get_logs_with_errors() and test_status), 'Test Failed'
```

Comment:
Maybe it would be more informative if it was in two asserts:
```
assert not get_logs_with_errors()
assert test_status
```
You would know instantly which one of those conditions failed. Although, I think that in case of a failure there will be printed output of those two variables so it might be ok.

Reply:
I put the together since i want both of them will be check.
if they will be in separate calls and the first one will assert, the second one will not be tested.

Reply:
Ok, that makes sense.